packages/js-sdk/src/sandbox.ts

```diff
@@ -170,149 +264,253 @@ export class Sandbox extends SandboxBase {
+    );
+  }
+
+  /**
+   * Move the mouse to the given coordinates.
+   * @param x - The x coordinate.
+   * @param y - The y coordinate.
+   */
+  async moveMouse(x: number, y: number): Promise<void> {
+    await this.commands.run(
+      `xdotool mousemove --sync ${x} ${y}`, { envs: { DISPLAY: this.display } }
+    );
+  }
+
+  /**
+   * Get the current cursor position.
+   * @returns A object with the x and y coordinates or null if the cursor is not visible.
+   */
+  async getCursorPosition(): Promise<CursorPosition | null> {
+    const result = await this.commands.run(
+      "xdotool getmouselocation", { envs: { DISPLAY: this.display } }
+    );
+
+    const match = result.stdout.match(/x:(\d+)\s+y:(\d+)/);
+    if (match) {
+      const [, x, y] = match;
+      if (x && y) {
+        return { x: parseInt(x), y: parseInt(y) };
+      }
+    }
+    return null;
+  }
+  /**
+   * Get the current screen size.
+   * @returns An {@link ScreenSize} object or null if the screen size is not visible.
+   */
+  async getScreenSize(): Promise<ScreenSize | null> {
+    const result = await this.commands.run(
+      "xrandr", { envs: { DISPLAY: this.display } }
+    );
+
+    const match = result.stdout.match(/(\d+x\d+)/);
+    if (match) {
+      const [width, height] = match[1].split('x').map(Number);
+      return { width, height };
+    }
+    return null;
+  }
+
+  private *breakIntoChunks(text: string, n: number): Generator<string> {
+    for (let i = 0; i < text.length; i += n) {
+      yield text.slice(i, i + n);
+    }
+  }
+
+  private quoteString(s: string): string {
+    if (!s) {
+      return "''";
+    }
+
+    if (!/[^\w@%+=:,./-]/.test(s)) {
+      return s;
+    }
+
+    // use single quotes, and put single quotes into double quotes
+    // the string $'b is then quoted as '$'"'"'b'
+    return "'" + s.replace(/'/g, "'\"'\"'") + "'";
   }
 
   /**
    * Write the given text at the current cursor position.
    * @param text - The text to write.
+   * @param chunkSize - The size of each chunk of text to write.
+   * @param delayInMs - The delay between each chunk of text.
    */
-  async write(text: string) {
-    return this.runPyautoguiCode(`pyautogui.write(${JSON.stringify(text)})`)
+  async write(
+    text: string,
+    chunkSize: number = 25,
+    delayInMs: number = 75
+  ): Promise<void> {
+    const chunks = this.breakIntoChunks(text, chunkSize);
+
+    for (const chunk of chunks) {
+      await this.commands.run(
+        `xdotool type --delay ${delayInMs} ${this.quoteString(chunk)}`,
+        { envs: { DISPLAY: this.display } }
+      );
+    }
   }
 
   /**
    * Press a key.
    * @param key - The key to press (e.g. "enter", "space", "backspace", etc.).
    */
-  async press(key: string) {
-    return this.runPyautoguiCode(`pyautogui.press(${JSON.stringify(key)})`)
+  async press(key: string): Promise<void> {
+    await this.commands.run(
+      `xdotool key ${key}`, { envs: { DISPLAY: this.display } }
+    );
   }
```

Comment:
I feel like this should be desktop.keyboard.something() but @mlejva might disagree